topbin/rocq.ml

```diff
@@ -29,6 +26,79 @@ let with_sibling_exe opts prog args =
   let argv = Array.of_list (prog :: args) in
   Rocqshim.exec_or_create_process prog argv
 
+type subcommand =
+  | Compile
+  | Repl
+  | ReplWithDrop
+  | NativePrecompile
+  | Check
+  | Votour
+  | PpMlg
+  | Dep
+  | Doc
+  | Wc
+  | Workmgr
+  | Tex
+  | Makefile
+  | Timelog2Html
+
+let subcommands = [
+  ("compile", "Compile a Rocq source file", Compile);
+  ("c", "Alias for compile", Compile);
+  ("repl", "Interactive read-eval-print loop", Repl);
+  ("top", "Alias for repl", Repl);
+  ("repl-with-drop", "repl with Drop command", ReplWithDrop);
+  ("top-with-drop", "Alias for repl-with-drop", ReplWithDrop);
+  ("native-precompile", "Preprocess a compiled Rocq file for use by native_compute", NativePrecompile);
+  ("check", "Check a compiled Rocq file (alias for command rocqchk)", Check);
+  ("votour", "Low level debugging of compiled files", Votour);
+  ("preprocess-mlg", "Preprocess Rocq grammar files (.mlg) to produce OCaml sources", PpMlg);
+  ("pp-mlg", "Alias for preprocess-mlg", PpMlg);
+  ("dep", "Print dependencies for compiling Rocq files", Dep);
+  ("doc", "Generate documentation from a Rocq source file", Doc);
+  ("wc", "Count lines of code in a Rocq source file", Wc);
+  ("workmgr", "Control the number of parallel workers used by Rocq", Workmgr);
+  ("tex", "Process Rocq code in a Latex document", Tex);
+  ("makefile", "Generate a Makefile to compile a Rocq project", Makefile);
+  ("timelog2html", "Combine timing information and a Rocq source file", Timelog2Html);
+]
+
+let print_usage fmt () =
```

Comment:
You may want to have a look at the `Format.pp_open_tbox` family of functions to do the tabulation and doc wrapping.

The table in https://github.com/LPCIC/elpi/blob/master/src/parser/README.md is printed by parser_config.ml using that kind of boxes.

Reply:
it works now, no need to get fancy